In [1]:
import pandas as pd

**What is the average departure delay for all flights?
What is the average arrival delay for all flights?
Is there more variability in departure delays or arrival delays?
What percentage of flights had a departure delay of more than 15 minutes?
What is the range of arrival delays among the flights?**

In [2]:
flights = pd.read_excel('C:\data analytics\pandas_docs/fligt.xlsx')
flights

Flight Number  Departure Delay (minutes)  Arrival Delay (minutes)
0            123                         15                       10
1            456                          0                        5
2            789                         20                       15
3             12                         10                       25
4            345                          5                        0

In [3]:

flights['Departure Delay (minutes)'].mean()

10.0

- 10 is the average deprture delay of all flights

In [5]:
flights['Arrival Delay (minutes)'].mean()

11.0

- 11 is the arrival delay of all flights

In [6]:
flights['Departure Delay (minutes)'].min()

0

In [7]:
flights['Departure Delay (minutes)'].max()

20

In [8]:
flights['Departure Delay (minutes)'].var()

62.5

In [9]:
(flights['Departure Delay (minutes)'].var())-(flights['Departure Delay (minutes)'].max())

42.5

In [10]:
flights['Arrival Delay (minutes)'].min()

0

In [11]:
flights['Arrival Delay (minutes)'].max()

25

In [12]:
flights['Arrival Delay (minutes)'].var()

92.5

In [13]:
(flights['Arrival Delay (minutes)'].var())-(flights['Arrival Delay (minutes)'].max())

67.5

- Here more variability in Arrival delay

In [14]:
(flights['Arrival Delay (minutes)'].max()) - (-flights['Arrival Delay (minutes)'].min())

25

- Range of the arrival delay is 25
- Range is the diffrnts b/w the largest and smallest values

In [15]:
flights[flights['Departure Delay (minutes)'] > 15]

Flight Number  Departure Delay (minutes)  Arrival Delay (minutes)
2            789                         20                       15

- flight number 789 had the delay of more than 15 min